onchain/rollups/test/foundry/portals/ERC20Portal.t.sol

```diff
@@ -1,408 +1,160 @@
-        uint256 aliceBalanceBefore = token.balanceOf(alice);
-        uint256 dappBalanceBefore = token.balanceOf(dapp);
-        uint256 portalBalanceBefore = token.balanceOf(address(portal));
-
-        // Expect InputAdded to be emitted with the right arguments
-        vm.expectEmit(true, true, false, true, address(inputBox));
-        emit InputAdded(dapp, 0, address(portal), input);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                Address.AddressEmptyCode.selector,
+                address(token)
+            )
+        );
 
-        // Transfer ERC-20 tokens to the DApp via the portal
+        vm.prank(alice);
         portal.depositERC20Tokens(token, dapp, _amount, _data);
-        vm.stopPrank();
-
-        // Check the balances after the deposit
-        assertEq(token.balanceOf(alice), aliceBalanceBefore - _amount);
-        assertEq(token.balanceOf(dapp), dappBalanceBefore + _amount);
-        assertEq(token.balanceOf(address(portal)), portalBalanceBefore);
-
-        // Check the DApp's input box
-        assertEq(inputBox.getNumberOfInputs(dapp), 1);
     }
 
-    function testRevertsOperationDidNotSucceed(
-        uint256 _amount,
-        bytes calldata _data
-    ) public {
-        // Create untransferable token
-        token = new UntransferableToken(alice, _amount);
-
-        vm.startPrank(alice);
-
-        token.approve(address(portal), _amount);
-
-        // Save the ERC-20 token balances
-        uint256 aliceBalanceBefore = token.balanceOf(alice);
-        uint256 dappBalanceBefore = token.balanceOf(dapp);
-        uint256 portalBalanceBefore = token.balanceOf(address(portal));
-
-        // Transfer ERC-20 tokens to the DApp via the portal
-        vm.expectRevert("SafeERC20: ERC20 operation did not succeed");
-        portal.depositERC20Tokens(token, dapp, _amount, _data);
-        vm.stopPrank();
-
-        // Same balances as before
-        assertEq(token.balanceOf(alice), aliceBalanceBefore);
-        assertEq(token.balanceOf(dapp), dappBalanceBefore);
-        assertEq(token.balanceOf(address(portal)), portalBalanceBefore);
-
-        // No input added
-        assertEq(inputBox.getNumberOfInputs(dapp), 0);
+    function testMockReturnsTrue(uint256 _amount, bytes calldata _data) public {
+        _testMockSuccess(_amount, _data, abi.encode(true));
     }
 
-    function testRevertsNonContract(
+    function testMockReturnsNothing(
         uint256 _amount,
         bytes calldata _data
     ) public {
-        // Use an EOA as token
-        token = IERC20(vm.addr(3));
-
-        vm.startPrank(alice);
-
-        // Transfer ERC-20 tokens to the DApp via the portal
-        vm.expectRevert("Address: call to non-contract");
-        portal.depositERC20Tokens(token, dapp, _amount, _data);
-        vm.stopPrank();
-
-        // No input added
-        assertEq(inputBox.getNumberOfInputs(dapp), 0);
+        _testMockSuccess(_amount, _data, abi.encode());
```

Comment:
They are equivalent. I chose `abi.encode()` because it's clearer that I am encoding something that might be ABI-decoded later. In this case, the return value of a function call. With `""`, you are assuming the reader knows that the empty return value (like `void` in C) is encoded as an empty byte array.